pkg/chunk/aws/s3_storage_client.go

```diff
@@ -47,6 +47,12 @@ func NewS3ObjectClient(cfg StorageConfig, schemaCfg chunk.SchemaConfig) (chunk.O
 		return nil, err
 	}
 
+	if cfg.S3ForceTLS {
+		if strings.Contains(cfg.S3.URL.Host, ".") {
+			s3Config = s3Config.WithEndpoint(fmt.Sprintf("https://%s", cfg.S3.URL.Host))
+		}
+	}
```

Comment:
I'd say this should be part of `awscommon.ConfigFromURL` - WDYT?

Reply:
ie I'm suggesting you should check the URL provided and preserve the scheme.

Reply:
Hm well given the example url `s3://abc:123@s3.default.svc.cluster.local:4569` would it just default to http then? 

So if it's https it's respected but all other schemes are http? This feels a little more obvious to me, but I don't disagree that it probably should be apart of `ConfigFromURL`.